oteapi/datacache/datacache.py

```diff
@@ -138,13 +138,10 @@ def __del__(self) -> None:
         self.diskcache.expire()
         self.diskcache.close()
 
-    def add(  # pylint: disable=too-many-arguments
+    def add(
         self,
         value: "Any",
-        key: "Optional[str]" = None,
-        expire: "Optional[int]" = None,
-        tag: "Optional[str]" = None,
-        json_encoder: "Optional[Any]" = None,
+        **options,
```

Comment:
What is the benefit replacing explicit keyword arguments with an opaque dict?

Reply:
I don't know if there are explicit benefits other than a minimization of the method signature, focusing it in on the most necessary an mostly used parameters.
It's "cleaner", but indeed, a bit more opaque.
This boils down to a "limit" of parameters set by pylint to 5, where these methods were going above 5.

I sort of understand why that limit is set, but at the same time it might be too low.

In the end, I think it's nice to focus the methods in in this way, relegating (un)important method options to a keyword argument.

Reply:
I am still not convinced. Hiding keyword arguments in an `**options` makes introspection impossible.
I also thing that the division between important and less important arguments is notion for humans, not machines. Therefore I think this division should be done in the docstring. This is also how numpy does it.

Reply:
Fair enough. I don't feel strongly in one way or another.

But let me ask you this then: How many parameters are then too many for a function/method? Because then I'll update the pylint configuration with this number - but it should also force up some thoughts around this subject.

Reply:
I need to push this through now, unfortunately, so I'll make an issue for this, revert the changes specific to this, and redo a disable comment for pylint here.